extensions/opentelemetry/opentelemetry-exporter-otlp-http/runtime/src/main/java/io/quarkus/opentelemetry/exporter/otlp/http/runtime/OtlpHttpExporterProvider.java

```diff
@@ -0,0 +1,16 @@
+package io.quarkus.opentelemetry.exporter.otlp.http.runtime;
+
+import javax.enterprise.inject.Produces;
+import javax.inject.Singleton;
+
+import io.quarkus.arc.DefaultBean;
+
+@Singleton
+public class OtlpHttpExporterProvider {
+    @Produces
+    @Singleton
+    @DefaultBean
+    public LateBoundBatchSpanProcessor batchSpanProcessorForOtlpHttp() {
+        return new LateBoundBatchSpanProcessor();
```

Comment:
Share this with the Jaeger extension.

Reply:
Right. When I move the class I will also move this Producer.

Reply:
On second thought, since this provider provides the bean as a singleton it looks like this will definitely cause problems when trying to use multiple exporters. Thus I am thinking it would possibly be simplest to leave it as-is in this PR and then try to refactor the code in a second step.

Reply:
Sure. Again, I'm not exactly sure how this will fit with the OTel AutoConfiguration extension.

Reply:
Are you referring to OTel auto-instrumentation? I was under the impression that this would be based on Java reflection and thus possibly be problematic for native mode and also not really Quarkus-idiomatic and further possibly also conflict with programmatic tracing performed by Quarkus.

Reply:
No. It's about configuring OTel: https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk-extensions/autoconfigure/README.md My idea was to support the exact same configuration options as detailed on OTel. This also means the support of custom exporters via ServiceLoader. 

As we have right now, I'm not totally sure that allowing exporters to be added as CDI beans is the right thing to do. I actually believe we have an issue over there because any discovered exported is added as a `SimpleSpanProcessor`, which is not really recommended: https://github.com/quarkusio/quarkus/blob/57f6a9d9cb288cce2b558d793948b8d0e56ea806/extensions/opentelemetry/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/TracerRecorder.java#L86-L89

Reply:
BTW, regarding instrumentation, I did a couple of changes to use the Instrumenter API (same used in the Java Agent). This provides a way more consistent experience between our implementation and the Java Agent: https://github.com/quarkusio/quarkus/pull/21583

Reply:
Yes, that makes sense. I will thus put this PR on hold until you have refactored the code for OpenTelemetry Autoconfigure.